### PR TITLE
Moved labels to enable parallelism pipeline in Luke model

### LINKED ISSUE #22561: Make all Transformer models compatible with model parallelism
Accelerate makes it easy to load a model on multiple GPUs with `device_map="auto"`. This in turn allows users to train model with naive model parallelism if they have several GPUs.

A problem that happens in Transformers, with model with heads (so not XxxModel but for instance XxxModelForSequenceClassification) is that the labels end up on a different device than the logits and there is a device mistmatch error.

Thankfully, there is an easy fix for that! #22535 shows how to fix this for T5 by just moving the labels to the same device as the logits they are compared to. This is a noop when the devices are the same, and fixes the issue if devices are different.

We would like help from the community to extend this to all models that support model parallelism, which are:
- [x] BART
- [x] BigBirdPegasus
- [x] BLIP2
- [x] BLOOM
- [x] BridgeTower
- [x] CamemBERT
- [ ] CLIP
- [ ] CLIPSeg
- [ ] CodeGen
- [x] Data2Vec Text
- [x] Deit
- [x] ESM
- [x] GPT-2
- [x] GPT-Neo
- [x] GPT-NeoX
- [x] GPT-NeoX Japanese
- [ ] GPT-J
- [x] GPT-San
- [ ] JukeBox
- [ ] Lilt
- [x] LLaMA (`LlamaForSequenceClassification` only)
- [x] Longformer
- [x] LongT5
- [ ] Luke
- [ ] M2M100
- [x] mBART
- [x] mT5
- [ ] NLLB
- [ ] OPT
- [ ] Owl-ViT
- [ ] Pix2Struct
- [x] PLBART
- [x] RoBERTa
- [x] RoBERTa PreLayerNorm
- [ ] SwitchTransformer
- [x] T5
- [ ] Vilt
- [x] ViT
- [x] ViT-Hybrid
- [ ] Whisper
- [x] XLM-RoBERTa

If you would like to grab one of those models and apply the same fix as #22535 to all the model with heads, please leave a comment here!

### PR DESCRIPTION
**Pull request by katiele47**
# What does this PR do?
Fixes #22561 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Please let me know if there's anything I need to correct! Thanks

@sgugger

### CODE REVIEW

**Comment by amyeroberts** [comment]
@katiele47  - thanks for the PR! 

Apologies, I reviewed another PR implementing the changes for Luke quickly this morning without properly reading the issue and realising this PR was also open. As the PR #22909 notes, there was just a small change that needed to happen on L2232, updating `logits` -> `reshaped_logits`. Otherwise the PR all looked good and after updating would have been merged :) 

I'm sorry for my mistake - I hope this doesn't discourage you from contributing and we welcome any PRs that you'd like to open in the future. 

@sushmanthreddy Anyone in the community is able to review PRs. If you spot something in the code that needs updating, could you comment directly on the PR instead of opening another one?

**Comment by sushmanthreddy** [comment]
@amyeroberts . I don't see, where I have gone wrong. I am new to open source and actually have worked on this issue before the @katiele47 did, but I haven't just mentioned a reviewer to review.
this is the proof for that [link](https://github.com/huggingface/transformers/pull/22900/files), actually, I have made the same changes needed just due to branch conflicts that haven't kept the proper pr.
anyways sorry for that if my mistake is there, I hope u understand

**Comment by katiele47** [comment]
Hi @amyeroberts thanks for spotting the small change and no worries! Now that it has been fixed by @sushmanthreddy PR #22909 should I close this PR?
